boto/gs/key.py

```diff
@@ -707,7 +713,12 @@ def set_contents_from_string(self, s, headers=None, replace=True,
         self.md5 = None
         self.base64md5 = None
 
-        fp = StringIO(get_utf8_value(s))
+        if six.PY3:
+            if isinstance(s, str):
+                s = s.encode('utf-8')
+            fp = io.BytesIO(s)
+        else:
+            fp = io.BytesIO(get_utf8_value(s))
```

Comment:
`io.BytesIO()` does not exist in Py2. We should change this to use `six.BytesIO`
Info: https://pythonhosted.org/six/#six.BytesIO

Reply:
Nice catch! Updated to six.BytesIO